### PR TITLE
DRIVERS-1675 Add the end session log expectations for CMAP

### PR DESCRIPTION
**Pull request by prestonvasquez**
DRIVERS-1675

Drivers need to account for the logs created by closing the client as an operation, including those produced by ending a session.

Please complete the following before merging:

- [ ] Update changelog. NA
- [x] Make sure there are generated JSON files from the YAML test files. 
- [x] Test changes in at least one language driver. Go
- [x] Test these changes against all server versions and topologies (including standalone, replica set, sharded clusters, and serverless).

### CODE REVIEW

**Comment by kmahar** [comment]
Thanks @prestonvasquez! This looks good to me based on what we discussed, but as I mentioned I can't prove that it works in Rust since we have a bug around not sending `endSession` which is non-trivial to fix and I won't be able to get to before my transfer. @BorisDog and I talked about him fixing the .NET bug and then using that to prove your change here works for another driver, so I'll defer to his review.

**Comment by kmahar** [comment]
BTW, I think it's ok to not file a separate drivers ticket for this since only Rust, Go, and .NET have gotten to it yet, but once this gets merged please update the downstream changes section on DRIVERS-1675 to say there was a follow up commit and link to the commit, since I had linked to my original commit there that didn't have this and people might sync to the old commit and miss this change otherwise.

File: source/connection-monitoring-and-pooling/tests/logging/connection-logging.yml
Comment (BorisDog) [inline source/connection-monitoring-and-pooling/tests/logging/connection-logging.yml:196]: +1

Comment (kmahar) [reply source/connection-monitoring-and-pooling/tests/logging/connection-logging.yml:196]: Somehow I missed this change when I originally reviewed the PR. What was the reasoning?

Comment (kmahar) [reply source/connection-monitoring-and-pooling/tests/logging/connection-logging.yml]: @BorisDog did you have some issue with this?

Comment (BorisDog) [reply source/connection-monitoring-and-pooling/tests/logging/connection-logging.yml]: No issues.
I thought the reason was missing, and was added now.

**Review by BorisDog — APPROVED** [review]
LGTM.
After `endSessions` in .NET the tests pass as expected.

**Review by kmahar — APPROVED** [review]
lgtm. since @BorisDog also owns CMAP and @patrickfreed is now on loan to service arch for the quarter I think it's ok to merge without waiting for Patrick's review.

**Pull request merged**
